packages/astro/src/core/config.ts

```diff
@@ -88,46 +130,33 @@ export const AstroConfigSchema = z.object({
 		})
 		.optional()
 		.default({}),
-	markdownOptions: z
-		.object({
-			render: z.any().optional().default(['@astrojs/markdown-remark', {}]),
-		})
-		.strict()
-		.optional()
-		.default({}),
-	buildOptions: z
+	markdown: z
 		.object({
-			site: z
-				.string()
+			drafts: z.boolean().optional().default(false),
+			mode: z
+				.union([z.literal('md'), z.literal('mdx')])
 				.optional()
-				.transform((val) => (val ? addTrailingSlash(val) : val)),
-			sitemapFilter: z.function().optional(),
-			sitemap: z.boolean().optional().default(true),
-			pageUrlFormat: z
-				.union([z.literal('file'), z.literal('directory')])
+				.default('md'),
+			syntaxHighlight: z
+				.union([z.literal('shiki'), z.literal('prism'), z.literal(false)])
 				.optional()
-				.default('directory'),
-			legacyBuild: z.boolean().optional().default(false),
-			experimentalStaticBuild: z.boolean().optional().default(true),
-			experimentalSsr: z.boolean().optional().default(false),
-			drafts: z.boolean().optional().default(false),
+				.default('shiki'),
+			// TODO: add better type checking
+			shikiConfig: z.any().optional().default({}),
+			remarkPlugins: z.array(z.any()).optional().default([]),
+			rehypePlugins: z.array(z.any()).optional().default([]),
```

Comment:
Our markdown integration [doesn't distinguish](https://github.com/withastro/astro/blob/23b8ee2738076225da57ce5e3826373c8964717c/packages/markdown/remark/src/index.ts#L59-L60) Remark from Rehype plugins, so maybe it's better to have them in a single `plugins` array. What do you think?

Reply:
I second this if possible!

Reply:
It actually does AFAICT. They're both loaded at the same time, but _when_ the plugins are applied is important. Remark plugins come first, then we convert Remark => Rehype, then we apply the Rehype plugins.